app/uk/gov/hmrc/agentsubscription/controllers/SubscriptionJourneyController.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscription.controllers
+
+import java.time.{ LocalDateTime, ZoneId, ZoneOffset }
+import java.util.UUID
+
+import com.google.inject.Inject
+import com.kenshoo.play.metrics.Metrics
+import play.api.libs.json.Json.toJson
+import play.api.libs.json.{ JsError, JsSuccess, JsValue }
+import play.api.mvc.{ Action, AnyContent, Result }
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+import uk.gov.hmrc.agentsubscription.auth.AuthActions
+import uk.gov.hmrc.agentsubscription.connectors.MicroserviceAuthConnector
+import uk.gov.hmrc.agentsubscription.model.InternalId
+import uk.gov.hmrc.agentsubscription.model.subscriptionJourney.SubscriptionJourneyRecord
+import uk.gov.hmrc.agentsubscription.repository.SubscriptionJourneyRepository
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+
+class SubscriptionJourneyController @Inject() (implicit
+  metrics: Metrics,
+  microserviceAuthConnector: MicroserviceAuthConnector,
+  subscriptionJourneyRepository: SubscriptionJourneyRepository,
+  ec: ExecutionContext)
+  extends AuthActions(metrics, microserviceAuthConnector) with BaseController {
+
+  private def localWithJsonBody(f: SubscriptionJourneyRecord => Future[Result], request: JsValue): Future[Result] =
+    Try(request.validate[SubscriptionJourneyRecord]) match {
```

Comment:
there should be one similar method available in `BaseController` or somewhere to read model from request, can we reuse it instead of creating this `private def` ?

Reply:
Yeah, maybe - I think this was probably just copied from somewhere else...  will take a look, thanks.